internal/controllers/user_devices_controller.go

```diff
@@ -133,6 +133,60 @@ func NewUserDevicesController(settings *config.Settings, dbs func() *db.ReaderWr
 	}
 }
 
+// DeviceOwnershipMiddleWare check that authenticated user owns the userDeviceID in the path or has a confirmed Eth address that owns the corresponding NFT
+func (udc *UserDevicesController) DeviceOwnershipMiddleWare(c *fiber.Ctx) error {
+	udi := c.Params("userDeviceID")
+	userID := helpers.GetUserID(c)
+
+	userDevice, err := models.UserDevices(
+		models.UserDeviceWhere.ID.EQ(udi),
+		models.UserDeviceWhere.UserID.EQ(userID),
+	).One(c.Context(), udc.DBS().Reader)
+	if err != nil && !errors.Is(err, sql.ErrNoRows) {
+		udc.log.Err(err).Msg("error while checking if authenticated user owns device or corresponding nft")
+		return err
```

Comment:
This will get logged, so you could think about removing the line above if it's not adding much. Could make this comment in a few places.